### ISSUE TITLE
[Wii] Add screen mode options for strect the screen respecting the original GB/GBC/GBA screen dimensions?

### ISSUE THREAD

**Issue by saulfabregwiivc** [mgba-emu/mgba#2860]
Hi there :)

This question could be confusing for some but is it possible to add options for this suggestion:
A screen mode like "Pixel Accurate" but that can be fit into any screen but respecting the original Game Boy/Game Boy Color/Game Boy Advance screen dimensions, not just 4:3 TVs but also 16:9 TVs.

**Current "Pixel-Accurate" setting**
on 4:3 TVs
GB/GBC
![image](https://user-images.githubusercontent.com/66485640/226206974-8d55b2ca-6724-4b76-a357-682ce260593a.png)
GBA
![image](https://user-images.githubusercontent.com/66485640/226207185-debe530c-ce97-49f2-a2c5-e0665fc701ca.png)

on 16:9 TVs
GB/GBC
![image](https://user-images.githubusercontent.com/66485640/226206793-45db4288-3362-4056-a112-cf773db3362a.png)
GBA
![image](https://user-images.githubusercontent.com/66485640/226207269-184e2b19-3e34-4fed-b9d2-70088c716be3.png)

Taken from mGBA standalone Wii port

**What i want you to consider for add a new "Pixel-Accurate (Adapted to TV)" setting**
on 4:3 TVs
GB/GBC
![image](https://user-images.githubusercontent.com/66485640/226207744-f12758ac-6a91-4c54-bffd-9cb154c1a35c.png)
GBA
![image](https://user-images.githubusercontent.com/66485640/226207768-05aee11e-cabe-4d54-aa85-aca4d18652e2.png)

on 16:9 TVs
GB/GBC
![image](https://user-images.githubusercontent.com/66485640/226207745-6f0a98da-5c71-4de9-bd72-d5928ffa0ac4.png)
GBA
![image](https://user-images.githubusercontent.com/66485640/226207772-7c5180a5-8652-4978-966b-0e70337984ec.png)

These last ones taken from RetroArch Wii port.

Would this be possible of get these screen modes on official mGBA Wii from RA Wii?

Not to mention your port runs pretty well optimized.

**Comment by endrift** [comment]
Those latter screenshots are very clearly not pixel-accurate--they're aspect ratio full-screen with no filtering, so you have disparate sized pixels. That should be something mGBA already supports, though it's not widescreen-aware afaik

**Comment by saulfabregwiivc** [comment]
> Those latter screenshots are very clearly not pixel-accurate--they're aspect ratio full-screen with no filtering, so you have disparate sized pixels. That should be something mGBA already supports, though it's not widescreen-aware afaik

You're right. These current modes on mGBA Wii are not 16:9 (widescreen) friendly.
Would you make widescreen friendly those modes?

**Comment by endrift** [comment]
I don't think you understand what "pixel accurate" means, or perhaps don't understand that wide-screen mode has non-square pixels. You cannot make a pixel accurate mode in widescreen that has the right aspect ratio as a result of the non-square pixels. The only actionable thing here is adding a widescreen aspect fit, but it won't be pixel accurate.

**Comment by saulfabregwiivc** [comment]
> The only actionable thing here is adding a widescreen aspect fit, but it won't be pixel accurate.

Well, if Pixel Accurate isn't possible with what i'm asking, then why not try this thing to mGBA Wii? (Widescreen aspect fit) :)

**Comment by endrift** [comment]
Yeah that's the action I'm planning on taking for this report.

**Comment by saulfabregwiivc** [comment]
Sorry for the question but...
Any update for adding a 16:9 (widescreen) aspect fit / 16:9 correction to mGBA Wii?

**Comment by saulfabregwiivc** [comment]
Hey there @endrift, it's me saulfabreg again.

I don't want to bother you but... are you thinking on a possible way for add 16:9 (widescreen) aspect fit / correction to mGBA Wii?

**Comment by endrift** [comment]
I've scheduled it for 0.11 so I don't lose track

**Comment by saulfabregwiivc** [comment]
Just tried the latest development build of mGBA on Wii, and these new 4:3/16:9 modes are working like a charm! Thanks a lot :)
